docs/api/sql/Function.md

```diff
@@ -2111,6 +2111,41 @@ Output:
 false
 ```
 
+## ST_IsValidDetail
+
+Introduction: Returns a row, containing a boolean `valid` stating if a geometry is valid, a string `reason` stating why it is invalid and a geometry `location` pointing out where it is invalid.
+
+This function is a combination of [ST_IsValid](#st_isvalid) and [ST_IsValidReason](#st_isvalidreason).
+
+The flags parameter is a bitfield with the following options:
+
+- 0 (default): Use usual OGC SFS (Simple Features Specification) validity semantics.
+- 1: "ESRI flag", Accepts certain self-touching rings as valid, which are considered invalid under OGC standards.
+
+Formats:
+
+```sql
+ST_IsValidDetail(geom: Geometry)
+```
+
+```sql
+ST_IsValidDetail(geom: Geometry, flag: Integer)
+```
+
+Since: `v1.6.1`
+
+SQL Example:
+
+```sql
+SELECT ST_IsValidDetail(ST_GeomFromWKT('POLYGON ((30 10, 40 40, 20 40, 30 10, 10 20, 30 10))'))
+```
+
+Output:
+
+```
+{false, Ring Self-intersection at or near point (30.0, 10.0, NaN), POINT (30 10)}
```

Comment:
The output should be 3 columns, not a json string.

Reply:
Where is the flink doc example?

Reply:
Will make the change. This function can't be ported to Flink as Flink doesn't support struct type, you made me aware of it, hence no docs for Flink.